src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -213,11 +213,22 @@ private async Task RegisterDesignerAttributeAsync(Document document, string desi
             var updateService = await GetUpdateServiceIfCpsProjectAsync(document.Project, cancellationToken).ConfigureAwait(false);
             if (updateService != null)
             {
-                // we track work by async token but doesn't explicitly wait for it. and update service is free-thread service,
-                // no need to switch to UI thread to use it
-                var asyncToken = _listener.BeginAsyncOperation("RegisterDesignerAttribute");
-                _ = updateService.SetProjectItemDesignerTypeAsync(document.FilePath, designerAttributeArgument)
-                                 .ReportNonFatalErrorAsync().CompletesAsyncOperation(asyncToken);
+                try
+                {
+                    // we track work by async token but doesn't explicitly wait for it. and update service is free-thread service,
+                    // no need to switch to UI thread to use it
+                    var asyncToken = _listener.BeginAsyncOperation("RegisterDesignerAttribute");
```

Comment:
Might want to pull the calls to the listener outside of this so you don't hide bugs if _its_ disposed?

Reply:
ah. right. thank you.